src/Editor.js

```diff
@@ -514,8 +515,36 @@ const Editor = () => {
                   true
                 );
               }}
+              onDoubleClick={() => setEditableListItemId(id)}
+              onBlur={() => setEditableListItemId("")}
             >
-              {name}
+              {editableListItemId === id ? (
+                <input
+                  type="text"
+                  autoFocus
+                  onChange={event => {
+                    event.persist();
```

Comment:
Pretty sure we won't need this because the value is getting saved in the `updatedName` variable below (we're not trying to access the `event` object again, which is the thing that get recycled):

```suggestion
                    event.persist();
```